test/utils.dart

```diff
@@ -175,47 +98,42 @@ StreamMatcher _collectStreamMatcher(block()) {
 /// it with [_collectStreamMatcher].
 ///
 /// [streamMatcher] can be a [StreamMatcher], a [Matcher], or a value.
-void _expectOrCollect(streamMatcher) {
+Future _expectOrCollect(streamMatcher) {
   if (_collectedStreamMatchers != null) {
-    _collectedStreamMatchers.add(new StreamMatcher.wrap(streamMatcher));
+    _collectedStreamMatchers.add(streamMatcher);
+    return null;
   } else {
-    _watcherEvents.expect(streamMatcher);
+    return expectLater(_watcherEvents, emits(streamMatcher));
   }
 }
 
 /// Expects that [matchers] will match emitted events in any order.
 ///
 /// [matchers] may be [Matcher]s or values, but not [StreamMatcher]s.
-void inAnyOrder(Iterable matchers) {
+Future inAnyOrder(Iterable matchers) async {
   matchers = matchers.toSet();
-  _expectOrCollect(nextValues(matchers.length, unorderedMatches(matchers)));
-}
-
-/// Expects that the expectations established in either [block1] or [block2]
-/// will match the emitted events.
-///
-/// If both blocks match, the one that consumed more events will be used.
-void allowEither(block1(), block2()) {
-  _expectOrCollect(
-      either(_collectStreamMatcher(block1), _collectStreamMatcher(block2)));
+  return _expectOrCollect(emitsInAnyOrder(matchers));
 }
 
 /// Allows the expectations established in [block] to match the emitted events.
 ///
 /// If the expectations in [block] don't match, no error will be raised and no
 /// events will be consumed. If this is used at the end of a test,
-/// [startClosingEventStream] should be called before it.
-void allowEvents(block()) {
-  _expectOrCollect(allow(_collectStreamMatcher(block)));
-}
+/// [pumpEventQueue] should be called before it.
+Future allowEvents(block()) =>
+    _expectOrCollect(mayEmit(_collectStreamMatcher(block)));
 
 /// Returns a matcher that matches a [WatchEvent] with the given [type] and
 /// [path].
-Matcher isWatchEvent(ChangeType type, String path) {
-  return predicate((e) {
-    return e is WatchEvent &&
-        e.type == type &&
-        e.path == p.join(_sandboxDir, p.normalize(path));
+StreamMatcher isWatchEvent(ChangeType type, String path) {
+  return new StreamMatcher((queue) async {
```

Comment:
This doesn't follow most of the requirements in the [`new StreamMatcher()` documentation](https://www.dartdocs.org/documentation/test/0.12.30/test/StreamMatcher/StreamMatcher.html).

Reply:
I opted to go with no description because the printed details are already clear. I have updated to ensure it doesn't consume a value if it isn't valid.

Reply:
The description also needs to be in the past tense.

> I opted to go with no description because the printed details are already clear.

I don't like this. Not having an error message means this won't compose well with other matchers.